### DISCUSSION TITLE
How do I run router.afterEach for every route?

### DISCUSSION THREAD

**Discussion by isaackearl** [nuxt/nuxt#19320] [Questions]
With vue-router I would write something like this:

```
router.afterEach((to, from, failure) => {
  if (failure) {
    return;
  }
  const store = useSiteInformationStore();
  store.lastRefresh = moment.utc();
});
```

in my router.ts file.  However in Nuxt 3 there are instead Middelware navigation guards which mimic the beforeEach functionality of vue-router... I'm wondering if there is anything in nuxt to run stuff AFTER navigating.

Ideally I wouldn't manually have to add something on every page, but could add an "after any page loads" type of middleware.

Any guidance is appreciated.

Thanks!

**Comment by Baroshem** [comment]
Hey,

What about using global route middleware? -> https://nuxt.com/docs/api/utils/add-route-middleware#global-route-middleware

And add your logic there?

**Reply by isaackearl** [reply]
From what I understand that global route middleware acts more as a navigation guard that runs before routing.  

Using that would be more of the equivalent of doing a beforeEach... I'm looking for how to do something AFTER every route.

**Reply by Baroshem** [reply]
In this case, I think there is no global configuration option that I am aware of based on the official docs -> https://nuxt.com/docs/api/composables/use-router#navigation-guards

**Comment by vanling** [comment]
I am currently also looking for a feature like this. Have to trigger some pageview event in an exotic external tracker.. my current solution is a dirty settimeout and just hope that page titles etc are set before it triggers. 

Let's see if i can use @Baroshem idea with useRouter and afterEach.

Idea? `middelware.global.afterEach.js` 😇

**Reply by vanling** [reply]
@isaackearl 

Maybe you can do something like this as a plugin?
```
export default defineNuxtPlugin((nuxt) => {
  useRouter().afterEach(() => {
    console.log('trigger')
  })
})
```

**Reply by isaackearl** [reply]
I think I tried this, but I'll do it again today and let you know how it goes.

**Reply by Baroshem** [reply]
@isaackearl have you tried it?

**Reply by gyhyfj** [reply]
I tried this plugin resolution, it works well, thank you all !
